src/barcodes/codabar/index.js

```diff
@@ -0,0 +1,56 @@
+// Encoding specification:
+// http://www.barcodeisland.com/codabar.phtml
+
+class codabar{
+	constructor(string){
+		this.string = string.toUpperCase();
+
+		if (this.string.search(/^[0-9\-\$\:\.\+\/]*$/) === 0) {
+			this.string = "A" + this.string + "A";
+		}
+
+		this.encodings = {
+			"0": "101010011",
+			"1": "101011001",
+			"2": "101001011",
+			"3": "110010101",
+			"4": "101101001",
+			"5": "110101001",
+			"6": "100101011",
+			"7": "100101101",
+			"8": "100110101",
+			"9": "110100101",
+			"-": "101001101",
+			"$": "101100101",
+			":": "1101011011",
+			"/": "1101101011",
+			".": "1101101101",
+			"+": "101100110011",
+			"A": "1011001001",
+			"B": "1010010011",
+			"C": "1001001011",
+			"D": "1010011001"
+		};
+	}
+
+	valid(){
+		return this.string.search(/^[A-D]?[0-9\-\$\:\.\+\/]*[A-D]$/) !== -1;
```

Comment:
I agree that it is a good way to implement it (to add it when it does not exist)

But the valid function can't be called before the object is created (and therefore the constructor called).
On line `8-10` you check if the start and stop characters are missing and add them in that case.
And here (line `37`) you only require the stop character to be present. (Should require both since you add them if not present)

This means that you will have the following results:

Input | .valid() |
------ | -------
`1234` | `true` 
`A1234B` | `true` 
`A1234` | `false` 
`1234A` | `true` 

Take a look at the last row. It should be false.